src/Nethermind/Nethermind.Evm/Tracing/GasEstimator.cs

```diff
@@ -60,15 +54,24 @@ public long Estimate(Transaction tx, BlockHeader header, EstimateGasTracer gasTr
             }
 
             // Execute binary search to find the optimal gas estimation.
-            return BinarySearchEstimate(leftBound, rightBound, rightBound, tx, header);
+            try
+            {
+                return BinarySearchEstimate(leftBound, rightBound, tx, header, cancellationToken);
+            }
+            catch (OperationCanceledException)
+            {
+                return 0;
```

Comment:
Don't we potentially have something estimated and can return a higher bound?

Reply:
The best thing we could return here is `rightBound`, since the gas estimation will be at most that. During the binary search we cannot tell for sure that the `mid` value is actually even close to a "correct" estimate.

In practice though I think it's a bad idea to catch this exception. I would rather let it propagate but I don't know how we're dealing with cancellations generally.

Reply:
Go super high and return maxvalue instead?

Reply:
Let's do that then.